testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/javascript/JavascriptAdapterTest.java

```diff
@@ -400,6 +398,58 @@ public void loginRequiredAction() {
                 .init(defaultArguments(), this::assertSuccessfullyLoggedIn);
     }
 
+    /**
+     * Test for scope handling via {@code initOptions}: <pre>{@code
+     * Keycloak keycloak = new Keycloak(); keycloak.init({.... scope: "profile email phone"})
+     * }</pre>
+     * See KEYCLOAK-14412
+     */
+    @Test
+    public void testScopeInInitOptionsShouldBeConsideredByLoginUrl() {
+
+        JSObjectBuilder initOptions = defaultArguments()
+                .loginRequiredOnLoad()
+                // phone is optional client scope
+                .add("scope", "profile email phone");
+
+        try {
+            testExecutor.init(initOptions);
+            // This throws exception because when JavascriptExecutor waits for AsyncScript to finish
+            // it is redirected to login page and executor gets no response
+
+            throw new RuntimeException("Probably the login-required OnLoad mode doesn't work, because testExecutor should fail with error that page was redirected.");
+        } catch (WebDriverException ex) {
+            // should happen
+        }
+
```

Comment:
You can put this here as well:
```suggestion
assertThat(jsDriver.getCurrentUrl(), containsString("&scope=openid%20profile%20email%20phone"));
```

Reply:
This doesn't work here:
```
Expected :a string containing "&scope=openid%20profile%20email%20phone"
Actual   :"https://localhost:8543/auth/realms/test/protocol/openid-connect/auth?client_id=account&redirect_uri=https%3A%2F%2Flocalhost%3A8543%2Fauth%2Frealms%2Ftest%2Faccount%2Flogin-redirect&state=0%2F4c2c6895-ca59-491c-b874-2d440ee84732&response_type=code&scope=openid"
```

The following test for the `phone` claim works though.

Reply:
Weird. This works for me, are you sure you used the exact line I pasted here?